recipes/opencl-headers/all/conanfile.py

```diff
@@ -31,7 +31,14 @@ def package(self):
     def package_info(self):
         self.cpp_info.filenames["cmake_find_package"] = "OpenCLHeaders"
         self.cpp_info.filenames["cmake_find_package_multi"] = "OpenCLHeaders"
+        self.cpp_info.set_property("cmake_file_name", "OpenCLHeaders")
+
         self.cpp_info.names["cmake_find_package"] = "OpenCL"
         self.cpp_info.names["cmake_find_package_multi"] = "OpenCL"
+        self.cpp_info.set_property("cmake_target_name", "OpenCL::Headers")
+        self.cpp_info.set_property("pkg_config_name", "OpenCL")
+
         self.cpp_info.components["_opencl-headers"].names["cmake_find_package"] = "Headers"
         self.cpp_info.components["_opencl-headers"].names["cmake_find_package_multi"] = "Headers"
+        self.cpp_info.components["_opencl-headers"].set_property("cmake_target_name", "OpenCL::Headers")
+        self.cpp_info.components["_opencl-headers"].set_property("pkg_config_name", "OpenCLHeaders")
```

Comment:
```suggestion
        self.cpp_info.components["_opencl-headers"].set_property("pkg_config_name", "OpenCLHeaders")
```

What?

Reply:
but this causes `PkgConfigDeps` to generate a self including file:
```
Name: OpenCL
Description: Conan package: OpenCL
Version: 2021.06.30
Requires: OpenCL
```

with this change it generates
```
Name: OpenCL
Description: Conan package: OpenCL
Version: 2021.06.30
Requires: OpenCLHeaders
```

and 
```
prefix=/home/conan/.conan/data/opencl-headers/2021.06.30/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
libdir1=${prefix}/lib
includedir1=${prefix}/include

Name: OpenCLHeaders
Description: Conan component: OpenCLHeaders
Version: 2021.06.30
Libs: -L"${libdir1}" -F Frameworks 
Cflags: -I"${includedir1}" 
```

Reply:
I would say it's a bug of `PkgConfigDeps`, it's still experimental.

Reply:
what would you expect the generator to do?

Reply:
Actually i would expect the component pkg-config file to override the global pkg-config file like it happens for `pkg-config` generator

Reply:
@SpaceIm, I think that here maybe the best would be not to declare the `pkg_config_name` property in the component at all, just for the root cpp_info, so Conan will create a new name for the component (`OpenCL-_opencl-headers.pc` in this case) that is included in the `OpenCL.pc` that is the one we want. 
Doing that names will not collide and also not declaring the property with other invented named we don't promote names that are not official... Also, for Conan 1.45 [we have added a warning to try to avoid names overlapping](https://github.com/conan-io/conan/pull/10263) when using `PkgConfigDeps`. WDYT?

Reply:
Conan 1.45 will bring the same behavior for `PkgConfigDeps`. It's already tested here https://github.com/conan-io/conan/pull/10344. The documentation is pending but it's ongoing.
In summary, you can put:

```python
self.cpp_info.components["_opencl-headers"].set_property("pkg_config_name", "OpenCL")
```
and you're not going to create the self including file anymore.